textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/neextrator/NamedEntityExtractor.java

```diff
@@ -72,7 +84,153 @@ public void open() throws Exception {
      */
     @Override
     public ITuple getNextTuple() throws Exception {
-        return null;
+        sourceTuple = sourceOperator.getNextTuple();
+        if (sourceTuple == null) {
+            return null;
+        } else {
+            if (returnSchema == null) {
+                returnSchema = Utils.createSpanSchema(sourceTuple.getSchema());
+            }
+            List<Span> spanList = new ArrayList<>();
+            for (Attribute attribute : searchInAttributes) {
+                String fieldName = attribute.getFieldName();
+                IField field = sourceTuple.getField(fieldName);
+                spanList.addAll(extractNESpans(field, fieldName));
+            }
+
+            ITuple returnTuple = Utils.getSpanTuple(sourceTuple.getFields(), spanList, returnSchema);
+            sourceTuple = sourceOperator.getNextTuple();
+            return returnTuple;
+        }
+    }
+
+    /**
+     * @param iField
+     * @return a List of spans of the extracted information
+     * @about This function takes an (TextField) IField and a String (the field's name) as input and uses the Stanford NLP package to process the field.
+     * It returns a list of spans
+     * <p>
+     * In the returning span: Value -> the word itself
+     * Key   -> NE_Constant
+     */
+    private List<Span> extractNESpans(IField iField, String fieldName) {
+        List<Span> spanList = new ArrayList<>();
```

Comment:
The code of this function is the core of this operator, and it needs good comments to explain.